trade/assets/js/index.js

```diff
@@ -0,0 +1,156 @@
+// index.js
+
+import { fetchCurrencies, fetchTimeSeries } from './service.js';
+import { showLoadingIndicator, hideLoadingIndicator, setFlag, setActiveTimeButton, displayDropdownList, calculateStartDate , currencyToFlagCode  } from './utils.js';
+
+document.addEventListener("DOMContentLoaded", () => {
+    let currencyChart;
+    const currency1Dropdown = document.getElementById('currency1');
+    const currency2Dropdown = document.getElementById('currency2');
+    const currencyNamesDiv = document.getElementById('currencyNames');
+    const ctx = document.getElementById('currencyChart').getContext('2d');
+    const timeControls = document.getElementById('timeControls');
+    const avgexchangerate = document.getElementById("price");
+    const percentage = document.getElementById("percentage");
+    const loadingIndicator = document.getElementById('loadingIndicator');
+    const flag1 = document.querySelector(".flag1");
+    const flag2 = document.querySelector(".flag2");
+    const storedCurrency1 = localStorage.getItem('selectedCurrency1');
+    const storedCurrency2 = localStorage.getItem('selectedCurrency2');
+    const storedLastPrice = localStorage.getItem('lastPrice');
+    const storedPercentageChange = localStorage.getItem('percentageChange');
+    const storedInterval = localStorage.getItem('selectedInterval') || '15m';
+    let currentTimeRange = storedInterval;
+
+    if (storedCurrency1) currency1Dropdown.value = storedCurrency1;
+    if (storedCurrency2) currency2Dropdown.value = storedCurrency2;
+
+    if (storedLastPrice && storedPercentageChange) {
+        percentage.innerHTML = `<span class="${storedPercentageChange < 0 ? 'text-danger' : 'text-success'}">${storedLastPrice} </span> <span class="${storedPercentageChange < 0 ? 'text-danger' : 'text-success'}">(${parseFloat(storedPercentageChange).toFixed(2)}%)</span>`;
+    }
+
+    setActiveTimeButton(storedInterval);
+
+    fetchCurrencies()
+        .then(response => {
+            const currencies = response.data.available_currencies;
+            hideLoadingIndicator(loadingIndicator);
+            displayDropdownList(currency1Dropdown, currencies, storedCurrency1 || 'EUR');
+            displayDropdownList(currency2Dropdown, currencies, storedCurrency2 || 'USD');
+            setFlag(currency1Dropdown.value, flag1);
+            setFlag(currency2Dropdown.value, flag2);
+            updateChart(currentTimeRange);
+        })
+        .catch(error => {
+            hideLoadingIndicator(loadingIndicator);
+            console.error('Error fetching the data:', error);
```

Comment:
We must show the user what happened, so display to the user the error (user-friendly message) 